peaseq-case.wdl

```diff
@@ -1,453 +1,576 @@
     }
 
     # collate all fastqfiles
-    Array[File] original_sample_R1 = flatten(select_all([sample_R1_srafile, sample_R1_fastqfiles]))
-    Array[File] original_sample_R2 = flatten(select_all([sample_R2_srafile, sample_R2_fastqfiles]))
-    Array[File] original_all_sample_fastqfiles = flatten(select_all([sample_R1_srafile, sample_R1_fastqfiles,sample_R2_srafile, sample_R2_fastqfiles]))
+    Array[File] original_sample_R1 = flatten(select_all([
+        sample_R1_srafile,
+        sample_R1_fastqfiles,
+    ]))
+    Array[File] original_sample_R2 = flatten(select_all([
+        sample_R2_srafile,
+        sample_R2_fastqfiles,
+    ]))
+    Array[File] original_all_sample_fastqfiles = flatten(select_all([
+        sample_R1_srafile,
+        sample_R1_fastqfiles,
+        sample_R2_srafile,
+        sample_R2_fastqfiles,
+    ]))
 
     # transpose to paired-end tuples
-    Array[Pair[File, File]] original_sample_fastqfiles = zip(original_sample_R1, original_sample_R2)
+    Array[Pair[File, File]] original_sample_fastqfiles = zip(original_sample_R1, original_sample_R2
+        )
 
-### ------------------------------------------------- ###
-### ---------------- S E C T I O N 1 ---------------- ###
-### ----------- B: remove Spike-IN reads ------------ ###
-### ------------------------------------------------- ###
+    ### ------------------------------------------------- ###
+    ### ---------------- S E C T I O N 1 ---------------- ###
+    ### ----------- B: remove Spike-IN reads ------------ ###
+    ### ------------------------------------------------- ###
 
     # if multiple fastqfiles are provided
     Boolean multi_fastqpair = if length(original_sample_fastqfiles) > 1 then true else false
     Boolean one_fastqpair = if length(original_sample_fastqfiles) == 1 then true else false
 
-    if ( defined(spikein_bowtie_index) || defined(spikein_reference) ) {
+    if (defined(spikein_bowtie_index) || defined(spikein_reference)) {
         scatter (eachfastq in original_all_sample_fastqfiles) {
-            call fastqc.fastqc as spikein_indv_fastqc {
-                input :
-                    inputfile=eachfastq,
-                    default_location=if multi_fastqpair then 'SAMPLE/' + sub(basename(eachfastq),'_R?[12]_....f.*q.gz|_R?[12].f.*q.gz','') + '/SpikeIn/FastQC' else if defined(results_name) then results_name + '/SpikeIn/FastQC' else sub(basename(eachfastq),'_R?[12]_....f.*q.gz|_R?[12].f.*q.gz','') + '/SpikeIn/FastQC'
+            call fastqc.fastqc as spikein_indv_fastqc { input:
+                inputfile = eachfastq,
+                default_location = if multi_fastqpair then "SAMPLE/" + sub(basename(
+                    eachfastq), "_R?[12]_....f.*q.gz|_R?[12].f.*q.gz", "") + "/SpikeIn/FastQC"
+                    else if defined(results_name) then "~{results_name + '/SpikeIn/FastQC'}"
+                    else sub(basename(eachfastq), "_R?[12]_....f.*q.gz|_R?[12].f.*q.gz", ""
+                    ) + "/SpikeIn/FastQC",
```

Comment:
1. this needs manual formatting
2. this _shouldn't_ work anywhere, as absolute paths shouldn't be accessible in the guest

Reply:
o. maybe this does evaluate to a relative path? Oi, it's hard to parse through this 🤦‍♀️

Reply:
It should be a relative path. It is quite difficult to parse through, but the arms start with `SAMPLE/`, `~{results_name}`, and the result of `basename(eachfastq)`. My preference is to remove all of the `default_location` logic and then the DNAnexus-specific reorg step should handle putting things in directories if necessary for that platform.